hydra-node/src/Hydra/Crypto.hs

```diff
@@ -121,24 +171,20 @@ instance FromJSON VerificationKey where
         (pure . HydraVerificationKey)
         . rawDeserialiseVerKeyDSIGN
 
-instance Arbitrary VerificationKey where
-  arbitrary = deriveVerificationKey . generateSigningKey <$> arbitrary
+instance HasTextEnvelope (VerificationKey HydraKey) where
+  textEnvelopeType _ =
+    "HydraVerificationKey_"
+      <> fromString (algorithmNameDSIGN (Proxy :: Proxy SignAlg))
 
--- | Serialise the verification key material as raw bytes.
-serialiseVerificationKeyToRawBytes :: VerificationKey -> ByteString
-serialiseVerificationKeyToRawBytes (HydraVerificationKey vk) = rawSerialiseVerKeyDSIGN vk
-
--- | Deserialise a verirfication key from raw bytes.
-deserialiseVerificationKeyFromRawBytes :: MonadFail m => ByteString -> m VerificationKey
-deserialiseVerificationKeyFromRawBytes bytes =
-  case rawDeserialiseVerKeyDSIGN bytes of
-    Nothing -> fail "failed to deserialise verification key"
-    Just key -> pure $ HydraVerificationKey key
-
--- | Get the Blake2b hash of a 'VerificationKey'.
-hashVerificationKey :: VerificationKey -> Hash Blake2b_256 VerificationKey
-hashVerificationKey (HydraVerificationKey vk) =
-  castHash $ hashVerKeyDSIGN vk
+-- | Create a new 'SigningKey' from a 'ByteString' seed. The created keys are
+-- not random and insecure, so don't use this in production code!
```

Comment:
Can we mark it as `{-# DEPRECATED #-}` ? I think this would prevent us from compiling test code with -WError but .. I see some obvious benefits. This is also enforceable by hlint I think (where we can whitelist some modules where a particular function is allowed) but that supposes we do all run and obey to hlint.

Reply:
It would probably trip the call site above in `deterministicSigningKey` (iirc)